mat/dense_arithmetic.go

```diff
@@ -758,17 +758,16 @@ func (m *Dense) Apply(fn func(i, j int, v float64) float64, a Matrix) {
 	}
 }
 
-// RankOne performs a rank-one update to the matrix a and stores the result
-// in the receiver. If a is zero, see Outer.
-//  m = a + alpha * x * y'
+// RankOne performs a rank-one update to the matrix a with the vectors x
+// and y, where x is treated as a column vector and y is treated as a row
+// vector. The result is stored in the receiver. If a is zero, see Outer.
+//  m = a + alpha * x * y
```

Comment:
For the mathematician in me this unadorned notation is confusing. I would preserve the mathematical notation here (only perhaps written with `^T` instead of `'`) and instead document that both vectors are considered to be (or treated as) column vectors regardless of their actual orientation.

Reply:
I don't have strong opinions either way, but I'll wait for @btracey to chime in.